projects/plugins/crm/src/automation/class-automation-workflow.php

```diff
@@ -248,18 +247,51 @@ public function add_trigger( string $string ) {
 	}
 
 	/**
-	 * Set Automation Logger
+	 * Set the automation engine
+	 *
+	 * @param Automation_Engine $engine An instance of the Automation_Engine class.
+	 * @return void
+	 */
+	public function set_engine( Automation_Engine $engine ) {
+		$this->automation_engine = $engine;
+	}
+
+	/**
+	 * Get the automation engine
+	 *
+	 * @return Automation_Engine
+	 *
+	 * @throws Workflow_Exception Throws an exception if there is no engine instance.
+	 */
+	protected function get_engine(): Automation_Engine {
+		if ( ! $this->automation_engine instanceof Automation_Engine ) {
+			throw new Workflow_Exception(
+				/* Translators: %s The ID of the workflow. */
+				sprintf( __( '[%s] Cannot run workflow logic without an engine instance', 'zero-bs-crm' ), $this->get_id() ),
```

Comment:
Should we translate the technical log?

Reply:
If the output is for the end user, I'd say yes, absolutely, but if it is for debugging purposes I'd keep it in English.

Reply:
Summary from 1:1 conversation: We're sticking with "machine friendly" messages (read: we won't translate them).
The UI that handles automations should provide guidance instead.

Reply:
We could create another task for removing the translation from any technical log output.